### DISCUSSION TITLE
Create Class, Interface and Enums as expanders in the TOC for .NET Documentation

### DISCUSSION THREAD

**Discussion by UmaHarano** [dotnet/docfx#9933] [General]
Hi,
The TOC for my .NET assembly namespace types lists "Classes" "Interface" and "Enums" as "Toc Titles". Is there a way to make them into expanders in the TOC instead? This will make the TOC more compact. I have many types in a namespace and it would nice to collapse them.

![image](https://github.com/dotnet/docfx/assets/6740984/a0f0b104-ebc6-4cc9-b731-f2d9c270c5c0)

I want this behavior instead:
![image](https://github.com/dotnet/docfx/assets/6740984/1e89502c-e4ea-4ba5-a682-741cf2f89d26)

**Comment by UmaHarano** [comment]
I am using the following attributes:


```json
"metadata": [
    {
      "src": [
        {
          "src": "src",
          "files": [
            "xxxxx.csproj"
          ]
        }
      ],
      "dest": "api/xxxx",      
      "memberLayout": "separatePages",
      "outputFormat": "apiPage"
    },
```

**Comment by filzrev** [comment]
Currently there is no options available to change TOC categories behaviors.

To change behaviors. It require source code changes.
So it might be better to create `[Feature Request] issue instead.

**Background**
TOC for ApiPage/Markdown is generated by [DotnetApiCatalog.Toc.cs](https://github.com/dotnet/docfx/blob/df66b7d4017bd9cff911b51c9ed1463ce1e854cc/src/Docfx.Dotnet/DotnetApiCatalog.Toc.cs).
And categories labels are inserted by at following lines.

https://github.com/dotnet/docfx/blob/df66b7d4017bd9cff911b51c9ed1463ce1e854cc/src/Docfx.Dotnet/DotnetApiCatalog.Toc.cs#L214-L244

**Example code to change behaviors**

I've confirmed it can display expandable categories.
By manually restructure TOC items.

![image](https://github.com/dotnet/docfx/assets/103790468/013e3350-b600-4079-a6ce-10299a5a6545)

**Code changes**
```
            void InsertCategory(TocNodeType type, string name)
            {
                if (items.All(i => i.type == type))
                    return;

                if (items.FirstOrDefault(i => i.type == type) is { } node)
                {
                    var head = new TocNode { name = name, items = items.Where(x => x.type == type).ToList() };
                    items.Insert(items.IndexOf(node), head);
                    items.RemoveAll(x => x.type == type);
                }
            }
```

**Comment by UmaHarano** [comment]
@filzrev 
Thank you!  I made a new FeatureRequest here: https://github.com/dotnet/docfx/issues/9934
